modules/authentication/components/ResetPassword.jsx

```diff
@@ -0,0 +1,69 @@
+import React, {
+  cloneElement,
+  Component,
+  PropTypes
+} from 'react';
+import mapFormValues from '../utils/mapFormValues';
+
+const propTypes = {
+  actions: PropTypes.shape({
+    authentication: PropTypes.shape({
+      resetPassword: PropTypes.func.isRequired
+    }).isRequired
+  }).isRequired,
+  children: PropTypes.node,
+  location: PropTypes.shape({
+    query: PropTypes.object.isRequired
+  }).isRequired,
+  router: PropTypes.shape({
+    replace: PropTypes.func.isRequired
+  })
+};
+
+class ResetPassword extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const values = mapFormValues(this.resetPasswordForm.elements);
+
+    const resetPasswordOptions = {
```

Comment:
If we're allowing additional children to be passed into this component, this really should be
```
const resetPasswordOptions = {
  ...values,
  'access-token': this.props.location.query.token,
  client: this.props.location.query.client_id,
  uid: this.props.location.query.uid
};
```